app/controllers/articles_controller.rb

```diff
@@ -19,9 +19,8 @@ def feed
                   @articles.where(featured: true).includes(:user)
                 end
 
-    unless @articles
-      render body: nil
-      return
+    unless @articles&.any?
+      not_found
```

Comment:
I'm curious why [this method](https://github.com/thepracticaldev/dev.to/blob/e9a0efbbf2aabc915723db49b2f064b8437783a1/app/controllers/application_controller.rb#L10-L12) `raise`s, but doesn't `render` with a `status: :not_found` here.

Asking this because when I looked at [these tests](https://github.com/thepracticaldev/dev.to/pull/6429/files#diff-51a8c61afb038e6fe94aa132cd0eda4cR15-R19), it seemed weird to me that we're testing that a `get "/feed"` request raises an error rather than testing that the `response.status` is `not_found` 🤔

Reply:
It's because Rails converts those exceptions in the corresponding HTTP status, but it only does it in production mode :D It's a little bit weird, I know 👀

Reply:
WHOAAAAA, TIL: `config.action_dispatch.rescue_responses`!!! never knew that was a thing.

so far in my career, i've always seen/use the `rescue_from ActiveRecord::RecordInvalid, with: : render_not_found` pattern, so the fact that actiondispatch has a `ExceptionWrapper` class is news to me 😂 

@rhymes can you point me to anywhere in the docs where they explain that it's only used in production? i can't find any instance of that being explained (why does it work different in dev/test?)

Reply:
I had the feeling it had something to do with `config.consider_all_requests_local` which is true in development but I dug a bit. I found the reason why in the Rails code and then circled back to the the [Configuring Rails](https://guides.rubyonrails.org/configuring.html) to make sure it was documented:

> ActionDispatch::ShowExceptions rescues any exception returned by the application and renders nice exception pages if the request is local or if config.consider_all_requests_local is set to true. If config.action_dispatch.show_exceptions is set to false, exceptions will be raised regardless.

This is where I ended up in Rails:

https://github.com/rails/rails/blob/e36d4a0381c522bf3904c2310d1bec99c763ff96/actionpack/lib/action_dispatch/middleware/debug_exceptions.rb#L27-L41

Reply:
This is really surprising: thanks for the deep dive @rhymes!

Reply:
I'm really curious about how this is all wired up in rails (smells like a blogggg possssssssst! 😂). Thanks for pointing me in the right direction, @rhymes !